From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aviana Cruz <gwencroft@proton.me>
Date: Fri, 23 Dec 2022 15:05:15 +0800
Subject: [PATCH] Implement [PandaSpigot-0092] Backport ArrowDMG fix by Aviana
 Cruz

Upstream commit: PaperMC/Paper@bb397ba74c034ffe5ae164d774a043cb7bca0472

diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
index 13d4a45773ccba0af6dbf849f388be5a186493df..64b210f66f29b4676560d57e79ac982cd82089b9 100644
--- a/src/main/java/net/minecraft/server/EntityLiving.java
+++ b/src/main/java/net/minecraft/server/EntityLiving.java
@@ -1294,13 +1294,26 @@ public abstract class EntityLiving extends Entity {
         return 0.42F;
     }
 
+    protected long lastJumpTime = 0L; // PandaSpigot - Backport ArrowDMG fix
     protected void bF() {
+        // PandaSpigot start - Backport ArrowDMG fix
+        long time = System.nanoTime();
+        boolean canCrit = true;
+        if (this instanceof EntityPlayer) {
+            canCrit = false;
+            if (time - this.lastJumpTime > (long)(0.250e9)) {
+                this.lastJumpTime = time;
+                canCrit = true;
+            }
+        }
+        // PandaSpigot end - Backport ArrowDMG fix
+
         this.motY = (double) this.bE();
         if (this.hasEffect(MobEffectList.JUMP)) {
             this.motY += (double) ((float) (this.getEffect(MobEffectList.JUMP).getAmplifier() + 1) * 0.1F);
         }
 
-        if (this.isSprinting()) {
+        if (canCrit && this.isSprinting()) { // PandaSpigot - Backport ArrowDMG fix
             float f = this.yaw * 0.017453292F;
 
             this.motX -= (double) (MathHelper.sin(f) * 0.2F);
